vendor/fknsrs.biz/p/ottoext/loop/loop.go

```diff
@@ -144,46 +144,22 @@ func (l *Loop) processTask(t Task) error {
 	return nil
 }
 
-// Run handles the task scheduling and finalisation. It will block until
-// there's no work left to do, or an error occurs.
+// Run handles the task scheduling and finalisation.
+// It runs infinitely waiting for new tasks.
 func (l *Loop) Run() error {
-	for {
-		l.lock.Lock()
-		if len(l.tasks) == 0 {
-			// prevent any more tasks entering the ready channel
-			l.closed = true
-
-			l.lock.Unlock()
-
-			break
-		}
-		l.lock.Unlock()
-
-		t := <-l.ready
-
-		if t != nil {
-			if err := l.processTask(t); err != nil {
-				return err
-			}
+	for t := range <-l.ready {
```

Comment:
Are you sure it works as expected? Shouldn't it be `for t := range l.ready`? Here is an example https://gobyexample.com/range-over-channels. With this code, it looks like `range` is applied to a value received from `l.ready` channel, not to the channel itself.

Reply:
@adambabik oh wow! It definitely is! Can't believe I did this mistake.
Great spot!